### DISCUSSION TITLE
Database mirroring

### DISCUSSION THREAD

**Discussion by st4ycool** [IacobIonut01/Gallery#199] [General]
Hello, lacoblonut01! 

I am about to commit to trying to repair v1-wip branch where you worked on mirroring mediastore in database. I think it's a pretty good idea, especially if in the future Media class will be extended with some user values that are absent in MediaStore. 

Before actually starting I want to confirm that I understand how it works now and how you see changes needed to complete this task.

1. So, now you are listening to MediaStore changes, and on each update in MediaStore you update flows. But for each screen that needs media, there is a new flow created for it in VM. So, what would be really cool is to subscribe to MediaResolver in one place - somewhere deep in Repository, and then each VM takes this flow and maps it to each screen's needs (Albums, Dates, etc)
2. As I see in v1-wip you kinda completely changed everything. How do you think, it's better to start from scratch or reuse what is already done in v1-wip?
3. What is your main working branch? before it was v1, but now it's main? I remember you added shimmering placeholders instead of "Empty media" screen, but in main branch I don't see it anymore

**Comment by IacobIonut01** [comment]
Firstly, thanks for you contribution.

About the first thing: would be cool to have one flow to rule them all, if you figure out a way to have them and filtered on demand while maintaining the same or better speed than directly querrying the mediastore, sure, let me know when you start working, I'll create another branch v2-tests to push the changes to.
2nd: We can use v1-wip as a reference, but it's not mandatory to use the same base, best if we start from scratch
3rd: The main working branch is 'main'. The shimmering is still in the code, both that and 'Empty Media', you can easily observe the shimmering inside Search panel

**Comment by st4ycool** [comment]
Thanks for your reply. 
I will just double check how exactly pagination is going to work with DB

p.s. for the branch - I can start in local one and then push, or you can create one. as you wish
